lib/UI/exercise/exercise_card.dart

```diff
@@ -32,6 +22,16 @@ class ExerciseCard extends StatefulWidget {
     required this.workoutData,
     required this.totalsWidget,
   });
+  final ValueKey key;
+  final String name;
+  Exercise exercise;
+  Function(Exercise) addExercise;
+  Function() updateExisitingExercise;
+  Function(Exercise) removeExercise;
+  Function(Exercise, Sets, int) removeSet;
+  final bool isTemplate;
+  WorkoutData workoutData;
+  ExerciseTotalsWidget totalsWidget;
```

Comment:
Maybe group the final from the non-final variables so it looks cleaner and give the functions a return type if it's in the scope of this PR.